tests/NHSD.BuyingCatalogue.Identity.Api.Testing.Data/Entities/DataProtectionKey.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Xml.Linq;
+
+namespace NHSD.BuyingCatalogue.Identity.Api.Testing.Data.Entities
+{
+    public sealed class DataProtectionKey : IEquatable<DataProtectionKey>
+    {
+        public DataProtectionKey(string xml)
+            : this(XElement.Parse(xml))
+        {
+        }
+
+        public DataProtectionKey(XElement key)
+        {
+            Element = key;
+            Id = key.Attribute("id")?.Value;
+            FriendlyName = $"key-{Id}";
+            Xml = key.ToString(SaveOptions.DisableFormatting);
+        }
+
+        public XElement Element { get; }
+
+        public string FriendlyName { get; }
+
+        public string Id { get; }
+
+        public string Xml { get; }
+
+        public bool Equals(DataProtectionKey other)
```

Comment:
One for the coding standards but I think it would be a good idea to use the same equality format across all objects that override the Equals method. I did it slightly different but I quite like the majority of the functionality in the `IEquatable` `Equals` method.

Reply:
I'll leave this as is for now then and we can discuss our preferred implementation pattern in our next standards meeting.

Reply:
👍